packages/perseus/src/widgets/radio/choice-icon/cross-out-line.jsx

```diff
@@ -26,6 +26,7 @@ function CrossOutLine(props: {color: string, sat?: boolean, ...}): React.Node {
             height={crossOutLineSize}
             viewBox={`0 0 ${crossOutLineSize} ${crossOutLineSize}`}
             className={css(styles.crossOutLine)}
+            data-test-id="cross-out-line"
```

Comment:
I don't really love adding test IDs; it's kind of an antipattern (although I acknowledge somewhere else you're using a test ID I added, but I didn't review my own PR 😅). My gut reaction:

- Is this test valuable enough to introduce an antipattern?
- If it's important, shouldn't it be "visible" to [screen readers](https://www.smashingmagazine.com/2021/05/accessible-svg-patterns-comparison/)? Does it need things like a role and a title which we could use instead?
- Is there another way to get it if we don't think it should be "visible" to screen readers? Like `getElementsByTagName("line")`?
- If we decide it's a valuable test, shouldn't be accessible, can't be grabbed except with a test ID, then I think we need a very specific test ID. I recommend using [BEM](https://getbem.com/introduction/), which this is following but I worry `cross-out-line` will be too generic.

Reply:
I don't think there were tests checking the functionality of the cross-out-line before, so I think it is valuable. That said, I think this feature is only used with sat products, though I am not 100% sure. If that's the case, it may become unnecessary in the future. But for now, I think it is worthwhile if we can make it work.

The content is visible to screen readers (SRs) via the SR specific radio input button I added on a previous PR. The a11y text states whether an option is crossed out or not, so the visual effect is for sighted users.

I did look into accessing an element by tag, and read a [StackOverflow](https://stackoverflow.com/questions/54234515/get-by-html-element-with-react-testing-library) saying that is unadvisable. I also saw in VS Code this message, "Avoid direct Node access. Prefer using the methods from Testing Library." That was from eslint(testing-library/no-node-access). Here's a more recent [StackOverflow](https://stackoverflow.com/questions/69143671/how-to-find-by-element-using-react-testing-library) that confirms you can't find by element in vanilla React Testing Library. 

Just wanted to document my thoughts and research with all those points above. I did change the test-id to hopefully be more specific, but let me know if it needs to be adjusted at all.

Reply:
This is fine as-is.

I just want to note that there really are no rules, we make the right decision for the right situation and we have linters for common mistakes. It's also usually some sort of trade-off: it might usually be wrong to get by tag, but it's also usually wrong to get by test ID.

In this particular case:
- RTL doesn't want you to test against implementation details. However in this case, the tag is basically the user-facing signal. A user does see a `div` or a `span`, but they do see a `line`. It's a visual symbol signaling information to the user, like text or a label. If we change a `div` to a `span` that doesn't change what we're saying to a user; if we change a `line` to a `circle` we are changing what we're saying to user.
- There's a good case to make that we don't want this visible to screen readers because of what you mentioned - there's already a11y text for that. Although you might argue that the "checked", "unchecked", etc a11y text would make more sense as label text on the symbol and would be better feature parity between sighted and non-sighted users.
- So given that we understand what the RTL linter is yelling at us for ("don't test implementation detail, test user interaction") and understanding that it's arguably inapplicable to this situation, it would be understandable to ignore the linting error.

My point isn't that this needs to change (we need to merge this PR and move onto bug squashing), it's that it's less important to follow linting rules and more important to understand what the rule is for and when it can be ignored.

Reply:
Cool! Thanks so much for that explanation. I definitely appreciate it :) It's good to keep this in mind.